src/algebraic_topology/fundamental_groupoid/retraction.lean

```diff
@@ -0,0 +1,117 @@
+/-
+Copyright (c) 2022 Mark Lavrentyev. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mark Lavrentyev
+-/
+import topology.homotopy.equiv
+import category_theory.equivalence
+import algebraic_topology.fundamental_groupoid.product
+
+/-!
+# Homotopic maps induce naturally isomorphic functors
+
+## Main definitions
+
+  - `is_retraction r` A continuous map `r : X → A` (where `A` is a subtype of `X`) is a retraction
+    when the restriction of r to A is the identity. Equivalently, composing r with the inclusion
+    `i : A → X` is just the identity map on A.
+
+  - `fundamental_groupoid_split_mono_of_top_inclusion` The induced map on `fundamental_groupoid X`
+    of the inclusion `i : A → X` is a split mono in the category Groupoid.
+  - `fundamental_groupoid_split_epi_of_top_retraction` The induced map on `fundamental_groupoid X`
+    of the retraction `r : X → A` is a split epi in the category Groupoid.
+-/
+
+noncomputable theory
+
+universe u
+
+open fundamental_groupoid
+open category_theory
+open fundamental_groupoid_functor
+
+open_locale fundamental_groupoid
+open_locale unit_interval
+
+
+def top_hom_of_continuous_map
+  {X Y : Type*} [topological_space X] [topological_space Y] (f : C(X, Y)) : X ⟶ Y := f
```

Comment:
You need to use `[Top.of](https://leanprover-community.github.io/mathlib_docs/topology/category/Top/basic.html#Top.of)` as in `Top.of X ⟶ Top.of Y`, otherwise you get a morphism in Type*, i.e. a function, instead of a morphism in Top.

Reply:
I believe this should be resolved with the changes I made to how stuff is defined.